PlayFabSDK/source/Qos/PlayFabQosApi.cs

```diff
@@ -3,22 +3,41 @@ namespace PlayFab.QoS
 {
     using EventsModels;
     using MultiplayerModels;
+    using System.Collections.Concurrent;
     using System.Collections.Generic;
+    using System.Linq;
     using System.Threading.Tasks;
 
     public class PlayFabQosApi
     {
+        private readonly PlayFabAuthenticationContext _authContext;
+        private const int DefaultPingsPerRegion = 10;
+        private const int DefaultDegreeOfParallelism = 4;
+        private const int NumTimeoutsForError = 3;
         private const int DefaultTimeoutMs = 250;
-        private readonly Dictionary<string, string> _dataCenterMap = new Dictionary<string, string>();
-        private readonly PlayFabMultiplayerInstanceAPI multiplayerApi = new PlayFabMultiplayerInstanceAPI(PlayFabSettings.staticPlayer);
-        private readonly PlayFabEventsInstanceAPI eventsApi = new PlayFabEventsInstanceAPI(PlayFabSettings.staticPlayer);
+        
+        private readonly PlayFabMultiplayerInstanceAPI multiplayerApi;
+        private readonly PlayFabEventsInstanceAPI eventsApi;
 
+        private Dictionary<string, string> _dataCenterMap;
+
+        public PlayFabQosApi(PlayFabApiSettings settings = null, PlayFabAuthenticationContext authContext = null)
+        {
+            _authContext = authContext ?? PlayFabSettings.staticPlayer;
+            
+            multiplayerApi = new PlayFabMultiplayerInstanceAPI(settings, _authContext);
+            eventsApi = new PlayFabEventsInstanceAPI(settings, _authContext);
```

Comment:
As feedback for me:
Did this seem like an intuitive way to set up these API objects?
Was it straightforward to figure out and use?

Reply:
Bump...

Reply:
Yea, it was fairly straightforward and intuitive. Very easy to figure out how to do what I wanted.  If I am going to be nitpicky, to be honest, I would have preferred not to have to new up a bunch of different objects. I think the swagger generated SDK is slightly easier to use because there is a single top level object where you can set things like auth parameters and then it has a bunch of objects hanging off of it that are the various APIs. It aids in discoverability as well as not requiring boilerplate code to pass around auth & settings context. See ThunderheadService.cs in the server solution and the usage of the "ControlPlane" class  (and IControlPlane interface) for an example.

---
In reply to: [374906222](https://github.com/PlayFab/CSharpSDK/pull/111#discussion_r374906222) [](ancestors = 374906222)

Reply:
That's not the first time I've heard that feedback about the single top level object.
I'm glad it's easy (enough) though.
Given that this feedback is independently produced multiple times, I will bring it up in a "someday" feature.
Thanks.